bundle/deploy/files/upload.go

```diff
@@ -25,6 +28,10 @@ func (m *upload) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 
 	err = sync.RunOnce(ctx)
 	if err != nil {
+		permissionError := filer.PermissionError{}
+		if errors.As(err, &permissionError) {
+			return permissions.ReportPermissionDenied(ctx, b, b.Config.Workspace.StatePath)
+		}
```

Comment:
The permission errors in `libs/filer/filer.go` all implement the `Is(other error)` function to allow equivalence with errors in the standard library. If you do the same you can rewrite this using `errors.Is(err, fs.ErrPermission)`.